rfc/0075-faster-failover-and-configuration-push.md

```diff
@@ -0,0 +1,315 @@
+have a body if the pushed version is older than the active configuration.
+
+The KV engine updates the pushed configuration version in the following cases:
+* Configuration sent to the SDK in response to a `GetClusterConfig` (`0xb5`) request.
+* Configuration pushed to the SDK that enabled the HELLO flag `ClustermapChangeNotification` (`0x0d`).
+
+Note, that `DedupeNotMyVbucketClustermap` affects `ClustermapChangeNotification` and `ClustermapChangeNotificationBrief`
+features, that described below. In other words, if deduplication enabled, the cluster configuration will be announce for
+the socket connection only once.
+
+### Enforcing Snappy Compression for Cluster Configuration Payloads
+
+[https://review.couchbase.org/c/kv_engine/+/192152]: #
+[https://review.couchbase.org/c/kv_engine/+/192316]: #
+
+The KV engine introduces a new HELLO flag called `SnappyEverywhere` with a value of `0x13`. Once this flag is
+negotiated, the node will always use the compressed version of the cluster configuration and data type flags will be set
+to `JSON | SNAPPY` (`0x03`).
+
+### `GetClusterConfig` and Out-of-Order Execution
+
+[https://issues.couchbase.com/browse/MB-56885]: #
+
+HELLO flag `UnorderedExecution` (`0x0e`) enables Out-of-Order (OoO) execution, so that the KV engine is being allowed to
+reorder operations. [kv\_engine/docs/UnorderedExecution.md][kv-unordered-execution] provides more details on this
+feature.
+
+The `GetClusterConfig` (`0xb5`) command is explicitly marked as compatible with OoO execution, allowing it to be served
+without waiting for the completion of in-flight operations. Specifically, `GetClusterConfig` will not wait for long
+operations such as mutations with SyncDurability requirements. All current SDKs are expected to be compatible with the
+OoO execution mode, so no changes are expected.
+
+### Cluster Configuration Notification Changes
+
+Prior to server version 7.6, the KV engine had an opt-in feature to push configuration updates to SDKs. This feature
+could be enabled using the HELLO flag `ClustermapChangeNotification` (`0x0d`), which depends on `Duplex` (`0x0c`). More
+details about `Duplex` can be found in [kv\_engine/docs/Duplex.md][kv-duplex]. When both flags are negotiated, the
+server will send unsolicited configuration updates to the SDK without expecting any acknowledgement mechanism. While
+this approach proves to have better responsiveness compared to [RFC-0024: Fast Failover][rfc-0024], it also has its own
+drawbacks, such as:
+
+1. The SDK subscribes all connections using HELLO, and during rebalance, all connections will receive all notifications.
+2. In a Lambda scenario, if failover occurs while the SDK process is paused, upon resuming, the SDK must process all
+   updates on all sockets. This process takes unnecessary time, unlike when the SDK polls every 2.5 seconds.
+
+Since version 7.6, the KV engine introduces the HELLO flag `ClustermapChangeNotificationBrief` (`0x1f`). This flag
```

Comment:
Since we have to send all the features at once in the HELLO flag, this implies that we have to request cluster notifications _and_ request the brief version, but on clusters that only support notifications without the brief, we will end up receiving huge payloads of configs.  Unless these flags are mutually exclusive?

Reply:
The flags are independent, and I will update paragraph to make it clear that we only send `ClustermapChangeNotificationBrief`. And in general when both flags supported by the KV engine, the brief version "wins".